### ISSUE TITLE
add inverse wishart cholesky

### ISSUE THREAD

**Issue by spinkney** [stan-dev/math#2714]
Add implementations of Wishart density and RNG with Cholesky factor parameters and variates.  Signatures exposed through Stan will be

```stan
/**
 * Return normalized density of `L_Sigma` so that `L_Sigma * L_Sigma'`
 * has an `inverse_wishart(nu, L_Omega * L_Omega')` distribution.
 */
real inverse_wishart_cholesky_lpdf(matrix L_Sigma, real nu, matrix L_Omega);
```

```stan
/**
 * Return unnormalized density of `L_Sigma` so that `L_Sigma * L_Sigma'`
 * has an `inverse_wishart(nu, L_Omega * L_Omega')` distribution.
 */
real inverse_wishart_cholesky_lupdf(matrix L_Sigma, real nu, matrix L_Omega);
```

```stan
/**
 * Return Cholesky factor of a random inverse Wishart draw with degrees of
 * freedom `nu` and scale `L_Omega * L_Omega'`.
 */
matrix inverse_wishart_cholesky_rng(real nu, matrix L_Omega);
```

**Comment by bob-carpenter** [comment]
@spinkney:  Our process is to create issues first, then name branches with the issue number.  I don't even know where this is documented, so it's not surprising new devs are confused.

I fixed this one for you and put it in the right form.  I can't rename your branch for compliance, though.

**Comment by bob-carpenter** [comment]
No worries, @spinkney---our dev-facing doc has been all over the place and we haven't been enforcing our intended process, so it's almost impossible to learn by example.  If you number things, then you can auto-close by adding a commit message that includes "fixes #2714".  Maybe you did that with the name somehow and that's how these got linked?

**Comment by rok-cesnovar** [comment]
It can be a commit message or text in the PR top post. So you can just write "fixed #x" or "closes #X" in the PR post as well (actually any of the words close, closes, fixes, fixed, resolve, resolves, ... count). I tend to use that as opposed to commit messages.

![image](https://user-images.githubusercontent.com/28476796/165350599-70de62b6-35a3-4ea9-b76b-d3eb316476ae.png)

You can also select the issue its supposed to be closing on the right-hand side of the PR text: 
![image](https://user-images.githubusercontent.com/28476796/165350791-fdc98893-8e39-4d45-87eb-3209cfa15bf8.png)
